app/screens/edit_profile/index.ts

```diff
@@ -23,8 +27,26 @@ const enhanced = withObservables([], ({database}: WithDatabaseArgs) => {
     const samlNicknameAttributeSet = observeConfigBooleanValue(database, 'SamlNicknameAttributeSet');
     const samlPositionAttributeSet = observeConfigBooleanValue(database, 'SamlPositionAttributeSet');
 
+    const rawCustomAttributes = currentUser.pipe(
+        switchMap((u) => (u ? observeCustomProfileAttributesByUserId(database, u.id) : of$([]))),
+    );
+
+    const customFields = observeCustomProfileFields(database);
+
+    // Convert attributes to the format expected by the component
+    const formattedCustomAttributes = combineLatest([rawCustomAttributes, customFields]).pipe(
+        switchMap(([attributes]) => {
+            if (!attributes?.length) {
+                return of$({} as CustomAttributeSet);
+            }
+
+            return of$(convertProfileAttributesToCustomAttributes(database, attributes, sortCustomProfileAttributes));
```

Comment:
0/5 Is it better to do this conversion here than handling both in the component?

Reply:
No especific reason, since I was removing the whole thing from the component I just did the conversion here so the component is a bit simpler. But I'm just moving complexity from one place to the other.
As a side-effect, testing the component is a bit easier.